validation/lifecycle.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"errors"
+	"io/ioutil"
+
+	tap "github.com/mndrix/tap-go"
+	rspec "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/opencontainers/runtime-tools/validation/util"
+)
+
+func main() {
+	t := tap.New()
+	t.Header(0)
+
+	prestart := rspec.Hook{
+		Path: "/usr/bin/prestart",
+		Args: []string{
+			"echo `pre-start called` >> $bundle/$rootfs/output",
+		},
+	}
+	poststart := rspec.Hook{
+		Path: "/usr/bin/poststart",
+		Args: []string{
+			"echo `post-start called` >> $bundle/$rootfs/output",
+		},
+	}
+	poststop := rspec.Hook{
+		Path: "/usr/sbin/poststop",
+		Args: []string{
+			"echo `post-stop called` >> $bundle/$rootfs/output",
+		},
+	}
+
+	g := util.GetDefaultGenerator()
+	g.AddPreStartHook(prestart)
+	g.AddPostStartHook(poststart)
+	g.AddPostStopHook(poststop)
+	g.SetProcessArgs([]string{"echo `process called` >> $bundle/$rootfs/output"})
+
+	var output = "$bundle/$rootfs/output"
+	config := util.LifecycleConfig{
+		Actions: util.LifecycleActionCreate | util.LifecycleActionStart | util.LifecycleActionDelete,
+		PreCreate: func(r *util.Runtime) error {
+			outputData, err := ioutil.ReadFile(output)
+			if err != nil {
+				return err
+			}
+			if string(outputData) != "" {
+				return errors.New("Wrong call")
```

Comment:
I am not quite sure what to say about this type of error. Any suggestions?

Reply:
> I am not quite sure what to say about this type of error.

The associated `specerror` entries start around [here][1].

[1]: https://github.com/opencontainers/runtime-tools/blob/eaecd9f609c4d68ec22e4737f6ecb0063cffc4b2/specerror/config.go#L69

Reply:
Depending on what is read, the specific error is different. It would be very troublesome to be accurate to every error condition. So instead of making a further distinction, I want to output the error directly.

Do you think it is necessary to distinguish between each error?

Reply:
Yes, each error should explicitly reflect the relevant 'specerror'.